src/gun.erl

```diff
@@ -261,10 +261,14 @@ check_options([Opt = {protocols, L}|Opts]) when is_list(L) ->
 		_ ->
 			{error, {options, Opt}}
 	end;
+check_options([{retry, infinity}|Opts]) ->
```

Comment:
That's unrelated and I'm not sure why you added this? A sufficiently large value would be good enough. And even if you want to *always* reconnect there's a point where Gun should give up because something is really wrong.